Sources/StreamChat/Workers/MessageUpdater_Tests.swift

```diff
@@ -129,15 +129,89 @@ final class MessageUpdater_Tests: XCTestCase {
             
             // Load the message
             let message = try XCTUnwrap(database.viewContext.message(id: messageId))
+            let extraData = try XCTUnwrap(message.extraData.map { try? JSONDecoder.default.decode([String:RawJSON].self, from: $0)} )
             
             // Assert `MessageEditing` error is received
             XCTAssertTrue(completionError is ClientError.MessageEditing)
             // Assert message stays in the same state
             XCTAssertEqual(message.localMessageState, state)
             // Assert message's text stays the same
             XCTAssertEqual(message.text, initialText)
+            // Assert message's extra data stays the same
+            XCTAssertEqual(extraData, [:])
         }
     }
+
+    func test_editMessage_updatesLocalMessageCorrectlyWithExtraData() throws {
+        let currentUserId: UserId = .unique
+        let messageId: MessageId = .unique
+        let updatedText: String = .unique
+        let extraData: [String: RawJSON] = ["custom" : .number(0)]
+        let updatedExtraData: [String: RawJSON] = ["custom" : .number(1)]
+
+        // Flush the database
+        try database.removeAllData()
```

Comment:
The db is empty at this moment so no need to flush it

Reply:
Done.